authentication/authentication.go

```diff
@@ -0,0 +1,61 @@
+package authentication
+
+import (
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/auth0/go-auth0/internal/client"
+)
+
+// Authentication is the auth client.
+type Authentication struct {
+	Database *Database
+	OAuth    *OAuth
+
+	url             *url.URL
+	basePath        string
+	debug           bool
+	http            *http.Client
+	auth0ClientInfo *client.Auth0ClientInfo
+	common          manager
+	clientID        string
+	clientSecret    string
+}
+
+type manager struct {
+	authentication *Authentication
+}
+
+// New creates an auth client.
+func New(domain string, options ...Option) (*Authentication, error) {
+	// Ignore the scheme if it was defined in the domain variable, then prefix
+	// with https as it's the only scheme supported by the Auth0 API.
+	if i := strings.Index(domain, "//"); i != -1 {
+		domain = domain[i+2:]
+	}
+	domain = "https://" + domain
```

Comment:
Could we simply expect the domain to be without scheme, instead of trying to get it into shape? Unless this is implemented this way in other Management SDKs

Reply:
This is the same as what the management SDK already does. From a quick look all SDKs have an expectation that it should be without the scheme but not all enforce it, which can lead to errors.

Java does [have some handling](https://github.com/auth0/auth0-java/blob/master/src/main/java/com/auth0/client/auth/AuthAPI.java#L148-L151), I _think_ [PHP does](https://github.com/auth0/auth0-PHP/blob/main/src/Configuration/SdkConfiguration.php#L1013-L1023), Node/Python don't touch the domain in any way